stablehlo/dialect/StablehloOps.td

```diff
@@ -2379,6 +2401,21 @@ def StableHLO_CustomCallOp: StableHLO_Op<"custom_call",
     custom<CustomCallTarget>($call_target_name) `(` $inputs `)`
       attr-dict `:` functional-type(operands, results)
   }];
+
+  let extraClassDeclaration = commonClassDeclaration # [{
+    mlir::Attribute buildDefaultValue() {
+      auto apiVersion = getApiVersion();
+      if (apiVersion == mlir::stablehlo::CustomCallApiVersion::API_VERSION_TYPED_FFI) {
+        return DictionaryAttr::get(getContext());
+      } else {
+        return StringAttr::get(getContext(), "");
+      }
+    }
+
+    mlir::Attribute getResolvedBackendConfig() {
+      return getBackendConfig().value_or(buildDefaultValue());
+    }
```

Comment:
Let's just declare the method in `extraClassDeclaration`, search for "CustomCall" in `StablehloOps.cpp` and implement it along with the other custom call impls there.

Also lets consolidate this to a single method called `getBackendConfigOrDefault` which has an impl like (untested code):

```
Attribute CustomCallOp::getBackendConfigOrDefault() {
  if (auto attr = getBackendConfigAttr()) return attr;
  if (getApiVersion() == CustomCallApiVersion::API_VERSION_TYPED_FFI) {
    return DictionaryAttr::get(getContext());
  }
  return StringAttr::get(getContext(), "");
}
```